scripts/filter_expected_warnings.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python
+"""Script to filter out old warnings from doxygen."""
+
+import sys
+import re
+
+
+class DoxygenWarning(object):
+    """Doxygen warning class."""
+
+    def __init__(self, firstline, filename, warning):
+        self.firstline = firstline
+        self.filename = filename
+        self.warning = warning
+        self.otherlines = []
+
+    def equals_ignoring_path_and_line_number(self, other):
+        """Return true if warnings have same filename and warning message."""
+        if self.filename != other.filename:
+            return False
+        if self.warning != other.warning:
+            return False
+        return self.otherlines == other.otherlines
+
+    def print_with_prefix(self, prefix):
+        print(prefix + self.firstline)
+        for line in self.otherlines:
+            print(prefix + '  ' + line)
+
+
+class WarningsList(object):
+    """List of Doxygen warnings."""
+
+    def __init__(self, all_lines):
+        """Create a list of the warnings in this file."""
+        self.warnings_list = []
+        current = None
+
+        warning_start_expr = re.compile(r'[^:]+/([^:/]+):\d+: (warning.*$)')
```

Comment:
A comment describing in prose, with example input/result for the regexp would be nice here... e.g. "Given an input line such as blah, this regexp will match and return foo in group(1), and wibble in group(2)